packages/teleport/src/Apps/AddApp/Automatically/Automatically.tsx

```diff
@@ -34,13 +34,15 @@ export default function Automatically(props: Props) {
 
   const [name, setName] = React.useState('');
   const [uri, setUri] = React.useState('');
+  const [showCmd, setShowCmd] = React.useState(false);
 
   function handleCreate(validator: Validator) {
     if (!validator.validate()) {
       return;
     }
 
     props.onCreate(name, uri);
+    setShowCmd(true);
```

Comment:
Hmmmm, I see, so `Generate Scripts` just uses the token that is already fetched and builds the command with the app name and URL, while `Regenerate` will fetch a new token, right?

I like it, but we still have one problem, the `cmd` is generated in the hook when, and it uses the app name and URL.

We can make a method in the hook, something like, `setCmdParams(appName, appUri)` that updates the `cmd` without fetching a new token. What do you think?

Reply:
hm, i think it's better, that we take `createAppBashCommand` out of the hook (so hook is only about fetching token), and move it closer to where it's really needed which is in `Automatic`

would this simplify things?

Reply:
Hmmm yes, I think it would 

We can make `Automatically` receives the token and generate the command when the form is submitted.

This way the hook only worries about fetching and storing the token and it's expiry date.

https://github.com/gravitational/webapps/pull/641/commits/a79399831d59b3e30e2c6dbaf3c407697ae30917 (there is a lot of changes, but mostly it is moving command methods and its tests to other files)

We still need a `useEffect` to listen to new token generations and create a new command:

```jsx
React.useEffect(() => {
    if (!name || !uri) {
      return;
    }

    const cmd = createAppBashCommand(token, name, uri);
    setCmd(cmd);
  }, [token]);
```

(added on a following commit, https://github.com/gravitational/webapps/pull/641/commits/e6875c1e2c3691b1c260539ccc10153e2b4c882c)

But I think this is much better

Please let me know what you think